### PR TITLE
Add unit test coverage for boottime_util_linux.go

### LINKED ISSUE #91864: Add unittest coverage for `boottime_util_linux.go`
**What would you like to be added**:
Currently, `pkg/kubelet/util/boottime_util_linux.go` does not have unit test coverage. Specifically, there's no unit test coverage for `GetBootTime`. I'd like to see unit test coverage added for `GetBootTime`. See `pkg/kubelet/util/util_unix_test.go` for a sample of how to write platform specific unit tests.

**Why is this needed**:
There's a fairly small chance of `GetBootTime` being modified, and thus breaking, in the future without this added test coverage. However, it should be relatively straightforward to add the test coverage, and could make a good first issue for a new contributor. There's also the added benefit that it helps keep the test coverage in `pkg/kubelet` high :) And we never know, it might help catch a bug some day :)

Extra note

I think this ticket could be a great candidate for someone who is new to
contributing to sig-node/k8s. **I'd prefer to please give first dibs for picking
this up to any folks who are currently underrepresented in the k8s community.** Please reach out to @mattjmcnaughton on k8s Slack (or post directly in #sig-node and tag me) if you're interested, and I'll assign you the issue. I'm happy to mentor and be helpful in whatever way I can :)

### PR DESCRIPTION
**Pull request by farah**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #91864

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
-->
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/util/bootime_util_linux_test.go
Comment (hongliang5316) [inline pkg/kubelet/util/bootime_util_linux_test.go:33]: I think this judgment should be in the first step

Comment (farah) [reply pkg/kubelet/util/bootime_util_linux_test.go:33]: I think style wise it probably should.

@hongliang5316 Any other ideas on how else you would you improve coverage on `GetBootTime`?

Comment (farah) [reply pkg/kubelet/util/bootime_util_linux_test.go:33]: Yeah I think style wise it probably should.

@hongliang5316 Any other ideas on how else you would you improve coverage on GetBootTime?

Comment (mattjmcnaughton) [reply pkg/kubelet/util/bootime_util_linux_test.go:33]: +1 I think that moving the nil check first is the most ideomatic.

**Review by mattjmcnaughton — COMMENTED** [review]
Awesome work @farah! Other than moving the nil check up first, I think this is excellent!

/ok-to-test

I think you will also need to run `hack/update-bazel.sh` to add this test file to `BUILD.bazel`.

After taking those steps, please ping me and I'll mark as lgtm and add someone with approval permissions to give a final review.

File: pkg/kubelet/util/bootime_util_linux_test.go
Comment (hongliang5316) [reply pkg/kubelet/util/bootime_util_linux_test.go:33]: @farah 
I think `GetBootTime` will often return `err == nil`, maybe you can mock `unix.Sysinfo` which inside `GetBootTime`  and make it return `err != nil` to improve coverage

**Comment by mattjmcnaughton** [comment]
On Thu, Jul 09, 2020 at 03:26:50AM -0700, hongliang wrote:
> @hongliang5316 commented on this pull request.
> 
> 
> 
> > +
> +package util
> +
> +import (
> +	"testing"
> +	"time"
> +)
> +
> +func TestGetBootTime(t *testing.T) {
> +	boottime, err := GetBootTime()
> +
> +	if !boottime.After(time.Time{}) {
> +		t.Errorf("Invalid system uptime")
> +	}
> +
> +	if err != nil {
> 
> @farah 
> I think `GetBootTime` will often return `err == nil`, maybe you can mock `unix.Sysinfo` which inside `GetBootTime`  and make it return `err != nil` to improve coverage 
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/92760#discussion_r452119509

I'd propose that for now we focus on just checking the correct error path and
potentially we can think about mocking `unix.SysInfo` in a future pr? We would
want to trade off the complexity added by using a mock w/ the benefits of the
greater test coverage.

**Comment by hongliang5316** [comment]
> On Thu, Jul 09, 2020 at 03:26:50AM -0700, hongliang wrote: @hongliang5316 commented on this pull request. > + +package util + +import ( + "testing" + "time" +) + +func TestGetBootTime(t *testing.T) { + boottime, err := GetBootTime() + + if !boottime.After(time.Time{}) { + t.Errorf("Invalid system uptime") + } + + if err != nil { @farah I think `GetBootTime` will often return `err == nil`, maybe you can mock `unix.Sysinfo` which inside `GetBootTime` and make it return `err != nil` to improve coverage -- You are receiving this because you commented. Reply to this email directly or view it on GitHub: [#92760 (comment)](https://github.com/kubernetes/kubernetes/pull/92760#discussion_r452119509)
> I'd propose that for now we focus on just checking the correct error path and potentially we can think about mocking `unix.SysInfo` in a future pr? We would want to trade off the complexity added by using a mock w/ the benefits of the greater test coverage.

@mattjmcnaughton 
LGTM!

**Review by mattjmcnaughton — APPROVED** [review]
/assign @derekwaynecarr

/lgtm

LGTM assuming the tests pass! 

Awesome first pr @farah :)

**Comment by mattjmcnaughton** [comment]
@farah both of these test failures look like flake to me. Once this diff is approved, the k8s-ci-robot will rerun the tests until the flakes stop.

/retest

**Comment by farah** [comment]
"Pending — Not mergeable. Must be in milestone v1.19."

Since 1.19 has been released, how can we update this?

**Comment by mattjmcnaughton** [comment]
> "Pending — Not mergeable. Must be in milestone v1.19."
> 
> Since 1.19 has been released, how can we update this?

Thanks for keeping tabs on this @farah :) AFAIK, k8s 1.19 has not been released yet - just a couple of release candidates. This message on the mailing list has some good info https://groups.google.com/d/msg/kubernetes-dev/4V5-ORCwoUM/chcyC-xnBAAJ

**Comment by farah** [comment]
oops, thanks for the clarification matt :)

**Comment by mattjmcnaughton** [comment]
On Sun, Aug 02, 2020 at 08:22:51AM -0700, Ali Farah wrote:
> oops, thanks for the clarification matt :)
> 
> -- 
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/92760#issuecomment-667687441

No worries - the release process can have a lot of moving pieces to keep track
off :)

**Comment by mattjmcnaughton** [comment]
On Sat, Aug 22, 2020 at 05:27:18AM -0700, Ali Farah wrote:
> hey @mattjmcnaughton, is it possible if we can now remove the v1.19 milestone requirement from the checks?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/92760#issuecomment-678634510

Hmmm, I'm pretty sure that once 1.19 is released, the master branch will open
back up, and this should get merged. I'm fairly confident that we shouldn't have
to make any manual changes (but not positive). Let's give this another week or
two, and if its not merged, let's sync up with someone who knows a bit more
about the CI tooling than I do :)

**Comment by farah** [comment]
@mattjmcnaughton thanks for the info. I hope I didn't come across as pestering you. I was just concerned about rebase issues and chasing approvals again but now I think of it it's unlikely for this PR as it's kind of isolated. Anyway happy to touch base again in a week or two.

**Comment by mattjmcnaughton** [comment]
On Wed, Aug 26, 2020 at 02:33:22AM -0700, Ali Farah wrote:
> @mattjmcnaughton thanks for the info. I hope I didn't come across as pestering you. I was just concerned about rebase issues and chasing approvals again but now I think of it it's unlikely for this PR as it's kind of isolated. Anyway happy to touch base again in a week or two. 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/92760#issuecomment-680771033

Not a problem at all :) Thanks for keeping track of this pr! I am +1 that
hopefully this pr is sufficiently isolated we won't run into many headaches w/
rebasing, etc...

**Pull request merged**
